### PR TITLE
Improve handling of `/Include/` statements

### PR DESCRIPTION
**Pull request by monkeydom**
- allows includes to work that are relative to the current file
- makes the language server return an URL to goto definition that works again

### CODE REVIEW

**Comment by monkeydom** [comment]
If you like this contribution then might I suggest a release? Being able to use your release binaries is helpful to us.

Also not sure if I need to fix the clippy as it is in unrelated existing code. Please advise.

File: ginko/src/dts/project.rs [hunk 1]
```diff
@@ -240,12 +240,12 @@ impl Project {
...
-        let mut parser = Parser::new(
-            lexer,
-            ParserContext {
-                include_paths: self.include_paths.clone(),
-            },
-        );
+        // add the file's directory to the include paths to allow local includes
+        let mut include_paths = self.include_paths.clone();
+        if let Some(parent) = file_name.parent() {
+            include_paths.insert(0, parent.into());
```
Comment (Schottkyc137) [inline ginko/src/dts/project.rs:246]: Is there a specific reason why you used `insert(0)`, and not `push`? The latter should generally be preferred unless there is a clear reason not to

**Review by Schottkyc137 — CHANGES_REQUESTED** [review]
I have updated the master with the clippy changes. Just merging it should work.
If time permits, I'll create a new release once your changes are merged.

File: ginko/src/dts/project.rs (hunk 1 above)
Comment (monkeydom) [reply ginko/src/dts/project.rs:246]: We want the path of the current file to be tried first. E.g. local files win over path files. Therefore it needs to be in position 0. Or am I missing something?

**Review by Schottkyc137 — APPROVED** [review]

**Comment by Schottkyc137** [comment]
Thanks of this improvement!

**Pull request merged**